common/Environments/Helpers/EnvironmentsNotificationHelper.cs

```diff
@@ -0,0 +1,244 @@
+        // them to be enable it or added again so we'll prompt them to reboot again.
+        if (_shouldShowHyperVRebootButton)
+        {
+            ShowRestartNotification();
+            return;
+        }
+
+        var userInAdminGroup = _windowsIdentityHelper.IsUserHyperVAdmin();
+        var featureEnabled = ManagementInfrastructureHelper.IsWindowsFeatureAvailable(CommonConstants.HyperVWindowsOptionalFeatureName) == FeatureAvailabilityKind.Enabled;
+
+        if (!featureEnabled && !userInAdminGroup)
+        {
+            // Notification to enable Hyper-V and add user to Admin group
+            _stackedNotificationsBehavior.ShowWithWindowExtension(
+                _microsoftHyperVText,
+                _stringResource.GetLocalized("HyperVAdminAddUserAndEnableHyperVMessage"),
+                InfoBarSeverity.Error,
+                AddUserToHyperVAdminGroupAndEnableHyperVCommand,
+                _stringResource.GetLocalized("HyperVAdminAddUserAndEnableHyperVButton"));
+        }
+        else if (!featureEnabled && userInAdminGroup)
+        {
+            // Notification to enable the Hyper-V feature when user is in the admin group
+            _stackedNotificationsBehavior.ShowWithWindowExtension(
+                _microsoftHyperVText,
+                _stringResource.GetLocalized("HyperVFeatureNotEnabledMessage"),
+                InfoBarSeverity.Error,
+                AddUserToHyperVAdminGroupAndEnableHyperVCommand,
+                _stringResource.GetLocalized("HyperVEnableButton"));
+        }
+        else if (featureEnabled && !userInAdminGroup)
+        {
+            // Notification to add user to the Hyper-V admin group when the feature is enabled
+            _stackedNotificationsBehavior.ShowWithWindowExtension(
+                _microsoftHyperVText,
+                _stringResource.GetLocalized("UserNotInHyperAdminGroupMessage"),
+                InfoBarSeverity.Error,
+                AddUserToHyperVAdminGroupAndEnableHyperVCommand,
+                _stringResource.GetLocalized("HyperVAdminAddUserButton"));
+        }
+    }
+
+    private void ShowErrorWithRebootAfterExecutionMessage(string errorMsg)
+    {
+        _stackedNotificationsBehavior.ShowWithWindowExtension(
+            _microsoftHyperVText,
+            errorMsg,
+            InfoBarSeverity.Error);
+    }
+
+    private void ShowRestartNotification()
+    {
+        _stackedNotificationsBehavior.ShowWithWindowExtension(
+            _microsoftHyperVText,
+            _stringResource.GetLocalized("RestartAfterChangesMessage"),
+            InfoBarSeverity.Warning,
+            RestartComputerCommand,
+            _stringResource.GetLocalized("RestartButton"));
+    }
+
+    [RelayCommand]
+    private void RestartComputer()
+    {
+        var startInfo = new ProcessStartInfo
+        {
+            WindowStyle = ProcessWindowStyle.Hidden,
+
+            // Restart the computer
+            FileName = Environment.SystemDirectory + "\\shutdown.exe",
```

Comment:
Should this be an instant restart, or should the computer restart after ~30 seconds?

Reply:
hmm I think instant is fine since there is no way to cancel it within those 30 seconds anyway

Reply:
There is a way to cancel it: "shutdown -a".